### DISCUSSION TITLE
Logging Interceptor is deprecated - what to use?

### DISCUSSION THREAD

**Discussion by lferna** [quarkiverse/quarkus-cxf#370] [Q&A] [answered]
Hi,
I want to set a Log Interceptor in Soap messages. I want to use:
LoggingInInterceptor and LoggingOutInterceptor, but both of them are deprecated:
_@deprecated use the logging module rt/features/logging instead_

Also, LoggingFeature is deprecated too.

I don't know If I can use in native mode this extension:
cxf-rt-features-logging

Or the best way is to create my own interceptor. 

Any ideas? Thanks.
Regards

**Comment by shumonsharif** [comment]
Hi @lferna You should be able to use something like the below with the `cxf-rt-features-logging` dependency, and I believe it should work in native mode, but please test and let me know if not.

```properties
# For clients
quarkus.cxf.client."fruitservice-client".features=org.apache.cxf.ext.logging.LoggingFeature

# For service endpoints
quarkus.cxf.endpoint."fruitservice".features=org.apache.cxf.ext.logging.LoggingFeature
```

For reference, here is additional documentation on `org.apache.cxf.ext.logging.LoggingFeature`:
https://cxf.apache.org/docs/message-logging.html

In case you need to set some properties on `LoggingFeature`, while there's several ways to go about it, simplest is possibly to create a `@Produces` method that returns a preconfigured `LoggingFeature` instance.

**Comment by lferna** [comment]
Thanks for your reply. In a first approximation, it works in local environment (jvm) but not in graalvm because it doesn't find the class. But I need to investigate further, I will use a method to configure some things and I will come back and give the results.
Thx

**Comment by shumonsharif** [comment]
Thanks, I created a new issue: #371 to track this.  We likely need to make a small change in the extension; please do keep us posted on your findings.

**Comment by shumonsharif** [comment, marked as answer]
Luis, I looked into this, but could not find any issue in the extension.  Just for reference, you can use something similar to the below (combined with the `application.properties` updates mentioned earlier) to leverage `cxf-rt-features-logging` in native mode.

```java
import io.quarkus.arc.Unremovable;
import io.quarkus.runtime.annotations.RegisterForReflection;
import org.apache.cxf.ext.logging.LoggingFeature;

import javax.enterprise.inject.Produces;

@RegisterForReflection(targets = {LoggingFeature.class})
public class Producers {

    @Produces
    @Unremovable
    LoggingFeature loggingFeature() {
        LoggingFeature loggingFeature = new LoggingFeature();
        loggingFeature.setPrettyLogging(true);
        return loggingFeature;
    }

}
```

**Comment by lferna** [comment]
Hi @shumonsharif ,
With your code snippet, it works now. This was the warning before:
"no such class org.apache.cxf.ext.logging.LoggingFeature".
I suppose it's because this class is not forced for reflection in native mode, could be?
Anyway, with your solution, it works fine!
Thank so much.
KR

**Reply by shumonsharif** [reply]
> I suppose it's because this class is not forced for reflection in native mode, could be?

Yes, that's correct.  Outside of the core CXF components, we currently leave it up to users to define what needs to be available in native mode.
